paper_framework_sources/p1897_initial_executor_algorithms_for_cpp23.md

```diff
@@ -419,66 +453,76 @@ int r = sync_wait(just(3));
 
 
 ### Wording
+
 The name `execution::sync_wait` denotes a customization point object.
-The expression `execution::sync_wait(S)` for some subexpression `S` is expression-equivalent to:
+For some subexpression `s` let `S` be a type such that `decltype((s))` is `S`.
+The expression `execution::sync_wait(s)` is expression-equivalent to:
 
- * `S.sync_wait()` if that expression is valid.
- * Otherwise, `sync_wait(S)`, if that expression is valid with overload resolution performed in a context that includes the declaration
+ * `s.sync_wait()` if that expression is valid and if `S` satisfies `typed_sender` and `S::value_types` is of length 0 or 1.
+ * Otherwise, `sync_wait(s)`, if that expression is valid, if `S` satisfies `typed_sender` and `S::value_types` is of length 0 or 1, with overload resolution performed in a context that includes the declaration
```

Comment:
I removed the value_types comment for now because of Eric's feedback about allowing explicitly providing the return type to the template. We can clarify this text later.